crates/modelardb_server/tests/integration_test.rs

```diff
@@ -797,6 +835,63 @@ fn ingest_time_series_and_flush_data(
     test_context.flush_data_to_disk();
 }
 
+/// Generates a time series with one tag and ingests it into a model table. Then a query is created
+/// from `segment_query` that cannot be rewritten so it is executed on segments, executes both,
+/// compares their query plans to ensure one query was rewritten and the other query was not, and
+/// then compares the query results to ensure they are within `error_bound`.
+fn assert_ne_query_plans_and_eq_result(segment_query: String, error_bound: f32) {
+    let mut test_context = TestContext::new();
+    let time_series = TestContext::generate_time_series_with_tag(false, None, Some("tag"));
+
+    ingest_time_series_and_flush_data(&mut test_context, &[time_series], TableType::ModelTable);
+
+    // The predicate will guarantee that all data points will be included in the query but which
+    // prevents the optimizer from rewriting the query due to its presence in segment_query.
+    let data_point_query = format!("{segment_query} WHERE timestamp >= 0::TIMESTAMP");
+
+    let data_point_query_plans = test_context
+        .execute_query(format!("EXPLAIN {}", data_point_query))
+        .unwrap();
+    let segment_query_plans = test_context
+        .execute_query(format!("EXPLAIN {}", segment_query))
+        .unwrap();
+
+    let data_point_query_plans_text = array!(data_point_query_plans, 1, StringArray);
+    let segment_query_plans_text = array!(segment_query_plans, 1, StringArray);
+    assert_ne!(data_point_query_plans, segment_query_plans);
+    assert!(data_point_query_plans_text.value(1).contains("GridExec"));
+    assert!(!segment_query_plans_text.value(1).contains("GridExec"));
+
+    let data_point_query_result_set = test_context.execute_query(data_point_query).unwrap();
+    let segment_query_result_set = test_context.execute_query(segment_query).unwrap();
+
+    if error_bound == 0.0 {
+        assert_eq!(data_point_query_result_set, segment_query_result_set);
+    } else {
+        assert_eq!(data_point_query_result_set.num_columns(), 1);
+        assert_eq!(data_point_query_result_set.num_rows(), 1);
+        assert_eq!(segment_query_result_set.num_columns(), 1);
+        assert_eq!(segment_query_result_set.num_rows(), 1);
+
+        let data_point_query_result = array!(data_point_query_result_set, 0, Float64Array);
+        let segment_query_result = array!(segment_query_result_set, 0, Float64Array);
+
+        let within_error_bound = modelardb_compression::is_value_within_error_bound(
+            ErrorBound::try_new(error_bound).unwrap(),
+            data_point_query_result.value(0) as f32,
+            segment_query_result.value(0) as f32,
+        );
+
+        assert!(
+            within_error_bound,
+            "{} is not within {}% of {}",
```

Comment:
Very minor but maybe add punctuation to the panic message.

Reply:
Fixed.